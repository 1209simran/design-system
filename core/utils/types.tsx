export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
