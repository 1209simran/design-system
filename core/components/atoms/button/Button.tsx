import * as React from 'react';
import Spinner from '@/components/atoms/spinner';
import Icon from '@/components/atoms/icon';
import classNames from 'classnames';

export type Appearance = 'basic' | 'primary' | 'success' | 'alert' | 'transparent';

export type Size = 'tiny' | 'regular' | 'large';

export type Alignment = 'left' | 'right';

export interface ButtonProps {
  /**
   * The size of `Button`
   * @default "regular"
   */
  size?: Size;
  /**
   * Color of the `Button`
   * @default "basic"
   */
  appearance?: Appearance;
  /**
   * Disables the `Button`, making it unable to be pressed
   */
  disabled?: boolean;
  /**
   * Expands the `Button` to full width
   */
  expanded?: boolean;
  /**
   * Adds loader inside `Button` when waiting for an action to complete
   */
  loading?: boolean;
  /**
   * Name of icon that is to be added inside `Button`
   * Material icon name
   */
  icon?: string;
  /**
   * Align icon left or right
   * @default "left"
   */
  iconAlign?: Alignment;
  /**
   * Text to be added inside `Button`
   */
  children?: string;
  /**
   * Specifies tab index of `Checkbox`
   */
  tabIndex?: number;
  /**
   * Adds className to `Card` component
   */
  className?: string;
  /**
   * Handler to be called when `Button` is clicked
   */
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  /**
   * Handler to be called when mouse pointer enters `Button`.
   */
  onMouseEnter?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  /**
   * Handler to be called when mouse pointer leaves `Button`.
   */
  onMouseLeave?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const sizeMapping = {
  tiny: 12,
  regular: 16,
  large: 20,
};

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
  const {
    appearance = 'basic',
    size = 'regular',
    iconAlign = 'left',
    tabIndex = 0,
    children,
    icon,
    expanded,
    loading,
    disabled,
    className,
    ...rest
  } = props;

  const buttonClass = classNames({
    ['Button']: true,
    ['Button--expanded']: expanded,
    [`Button--${size}`]: size,
    ['Button--square']: !children,
    [`Button--${appearance}`]: appearance,
    [`Button--iconAlign-${iconAlign}`]: children && iconAlign,
    [`${className}`]: className
  });

  const iconClass = classNames({
    ['Button-icon']: true,
    [`Button-icon--${iconAlign}`]: children && iconAlign
  });

  const spinnerClass = classNames({
    ['Button-spinner']: true,
    [`Button-spinner--${iconAlign}`]: children && iconAlign
  });

  return (
    <button ref={ref} className={buttonClass} disabled={disabled || loading} tabIndex={tabIndex} {...rest} >
      {loading && (
        <span className={spinnerClass}>
          <Spinner size="small" appearance={(appearance === 'basic' || appearance === 'transparent') ? 'secondary' : 'white'} />
        </span>
      )}
      {icon && !loading && (
        <div className={iconClass}>
          <Icon
            name={icon}
            appearance={disabled ? 'disabled' : (appearance === 'basic' || appearance === 'transparent') ? 'default' : 'white'}
            size={sizeMapping[size]}
          />
        </div>
      )}
      {children && `${children.charAt(0).toUpperCase()}${children.slice(1)}`}
    </button>
  );
});

Button.displayName = 'Button';

export default Button;
