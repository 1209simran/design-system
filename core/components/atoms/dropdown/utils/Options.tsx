export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
