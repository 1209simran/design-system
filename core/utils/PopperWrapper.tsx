import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Manager, Reference, Popper } from 'react-popper';

type PositionType =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start'
  | undefined;

type actionType = 'click' | 'hover';

enum Offsets {
  Small = '2px',
  Medium = '4px',
  Large = '8px'
}

interface IProps {
  trigger: React.ReactElement<any>;
  placement: PositionType;
  children: React.ReactElement<any>;
  style: React.CSSProperties;
  appendToBody: boolean;
  on?: actionType;
  offset: keyof typeof Offsets;
  closeOnBackdropClick?: boolean;
  hoverable?: boolean;
}

interface IState {
  open: boolean;
  mouseLeaveDelay: number;
  mouseEnterDelay: number;
}

class PopperWrapper extends React.Component<IProps, IState> {
  private triggerRef: React.RefObject<HTMLElement>;
  private popupRef: React.RefObject<HTMLDivElement>;
  private _timer?: number;

  constructor(props: IProps) {
    super(props);

    this.state = {
      open: false,
      mouseLeaveDelay: 50,
      mouseEnterDelay: 0
    };

    this.triggerRef = React.createRef();
    this.popupRef = React.createRef();
  }

  public componentWillUnmount() {
    clearTimeout(this._timer);
    document.removeEventListener('mousedown', this.doesNodeContainClick);
  }

  public handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: false });
        if (this.props.children.props.onMouseLeave) {
          this.props.children.props.onMouseLeave(event);
        }
      }, this.state.mouseLeaveDelay);
    } else {
      this.setState({ open: false });
      if (this.props.children.props.onMouseLeave) {
        this.props.children.props.onMouseLeave(event);
      }
    }
  }

  public handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: true });
        if (this.props.children.props.onMouseEnter) {
          this.props.children.props.onMouseEnter(event);
        }
      }, this.state.mouseEnterDelay);
    } else {
      this.setState({ open: true });
      if (this.props.children.props.onMouseEnter) {
        this.props.children.props.onMouseEnter(event);
      }
    }
  }

  public togglePopper = () => {
    this.setState({ open: !this.state.open });
  }

  public doesNodeContainClick = (event: Event) => {
    if (
      !(
        this.findDOMNode(this.popupRef).contains(event.target as HTMLElement) ||
        this.findDOMNode(this.triggerRef).contains(event.target as HTMLElement)
      )
    ) {
      this.togglePopper();
    }
  }

  public componentDidMount() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public componentDidUpdate() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    } else if (on === 'click' && !open && closeOnBackdropClick) {
      document.removeEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public getTriggerElement(trigger: React.ReactElement<any>, ref: React.Ref<any>, on: actionType) {
    const options =
      on === 'hover'
        ? {
          ref,
          onMouseEnter: this.handleMouseEnter,
          onMouseLeave: this.handleMouseLeave
        }
        : {
          ref,
          onClick: this.togglePopper
        };

    const element = React.cloneElement(
      (
        <span className="d-inline-block">
          {trigger}
        </span>
      ),
      options
      );

    return element;
  }

  public getChildrenElement(
    children: React.ReactElement<any>,
    ref: React.Ref<any>,
    placement: string,
    style: React.CSSProperties
  ) {
    const options = this.props.on === 'hover' ? {
      ref,
      style,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
      'data-placement': placement
    } : {
      ref,
      style,
      'data-placement': placement
    };

    const element = React.cloneElement(children, options);
    return element;
  }

  public render() {
    const { trigger, children, placement, appendToBody, on = 'click', offset } = this.props;
    const { open } = this.state;

    return (
      <Manager>
        <Reference innerRef={this.triggerRef}>{({ ref }) => this.getTriggerElement(trigger, ref, on)}</Reference>
        {open &&
          appendToBody &&
          ReactDOM.createPortal(
            (
              /* tslint:disable:no-shadowed-variable */
              <Popper placement={placement} innerRef={this.popupRef}>
                {({ ref, style, placement }) => {
                  const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
                  return this.getChildrenElement(children, ref, placement, newStyle);
                }}
              </Popper>
            ),
            document.body
          )}
        {open && !appendToBody && (
          <Popper placement={placement} innerRef={this.popupRef}>
            {({ ref, style, placement }) => {
              const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
              return this.getChildrenElement(children, ref, placement, newStyle);
            }}
          </Popper>
        )}
      </Manager>
    );
  }

  private getUpdatedStyle = (
    oldStyle: React.CSSProperties,
    placement: PositionType,
    offset: keyof typeof Offsets = 'Medium'
  ) => {
    const { style = {} } = this.props;
    const newStyle = { ...style, ...oldStyle };
    const position = placement ? placement.split('-')[0] : placement;
    switch (position) {
      case 'top':
        newStyle.marginBottom = Offsets[offset];
        break;

      case 'bottom':
        newStyle.marginTop = Offsets[offset];
        break;

      case 'left':
        newStyle.marginRight = Offsets[offset];
        break;

      case 'right':
        newStyle.marginLeft = Offsets[offset];
        break;
    }
    return newStyle;
  }

  private findDOMNode = (ref: React.RefObject<HTMLElement>) => {
    return ReactDOM.findDOMNode(ref.current!) as HTMLElement;
  }
}

export default PopperWrapper;
