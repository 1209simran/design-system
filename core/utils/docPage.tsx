// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
