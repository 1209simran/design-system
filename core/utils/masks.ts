import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
