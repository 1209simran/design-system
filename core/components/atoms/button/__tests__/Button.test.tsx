import * as React from 'react';
import { shallow } from 'enzyme';
import { TestHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/TestHelper';
import Button, { IButtonProps as IProps } from '../Button';

const BooleanValue = [true, false];
const Icon = ['events'];
const IconAlign = ['left', 'right'];
const Size = ['tiny', 'regular', 'large'];
const Appearance = ['basic', 'primary', 'success', 'alert', 'transparent'];
const FunctionValue = [jest.fn()];
const StringValue = ['button'];

const Mapper: Record<string, any> = {
  children: valueHelper(StringValue, { iterate: true }),
  size: valueHelper(Size, { iterate: true }),
  appearance: valueHelper(Appearance, { iterate: true }),
  expanded: valueHelper(BooleanValue, { iterate: true }),
  disabled: valueHelper(BooleanValue, { iterate: true }),
  loading: valueHelper(BooleanValue, { iterate: true }),
  icon: valueHelper(Icon, { iterate: true }),
  iconAlign: valueHelper(IconAlign, { iterate: true }),
  onClick: valueHelper(FunctionValue, { iterate: true }),
  onMouseEnter: valueHelper(FunctionValue, { iterate: true }),
  onMouseLeave: valueHelper(FunctionValue, { iterate: true }),
};

describe('Button component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;
    const { children, ...rest } = attr;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Button
          {...rest}
        >
          {children}
        </Button >
      );
      expect(tree).toMatchSnapshot();
    });
  };

  TestHelper(Mapper, testFunc);
});
