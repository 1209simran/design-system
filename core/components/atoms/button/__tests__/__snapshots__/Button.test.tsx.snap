// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component 
  appearance: "alert", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
  disabled={true}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--alert Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "alert", loading: true
 1`] = `
<button
  className="Button Button--regular Button--alert Button--iconAlign-left"
  disabled={true}
>
  <Spinner
    appearance="white"
    size="small"
  />
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  disabled={true}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--basic Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "basic", loading: true
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
  disabled={true}
>
  <Spinner
    appearance="secondary"
    size="small"
  />
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
  disabled={true}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--primary Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "primary", loading: true
 1`] = `
<button
  className="Button Button--regular Button--primary Button--iconAlign-left"
  disabled={true}
>
  <Spinner
    appearance="white"
    size="small"
  />
  Button
</button>
`;

exports[`Button component 
  appearance: "success", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
  disabled={true}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--success Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "success", loading: true
 1`] = `
<button
  className="Button Button--regular Button--success Button--iconAlign-left"
  disabled={true}
>
  <Spinner
    appearance="white"
    size="small"
  />
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", disabled: true
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--iconAlign-left"
  disabled={true}
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", expanded: false
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", expanded: true
 1`] = `
<button
  className="Button Button--expanded Button--regular Button--transparent Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  appearance: "transparent", loading: true
 1`] = `
<button
  className="Button Button--regular Button--transparent Button--iconAlign-left"
  disabled={true}
>
  <Spinner
    appearance="secondary"
    size="small"
  />
  Button
</button>
`;

exports[`Button component 
  size: "large"
 1`] = `
<button
  className="Button Button--large Button--basic Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  size: "large", icon: "events", iconAlign: "left"
 1`] = `
<button
  className="Button Button--large Button--basic Button--iconAlign-left"
>
  <div
    className="Button-icon Button-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={20}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "large", icon: "events", iconAlign: "right"
 1`] = `
<button
  className="Button Button--large Button--basic Button--iconAlign-right"
>
  <div
    className="Button-icon Button-icon--right"
  >
    <Icon
      appearance="default"
      name="events"
      size={20}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "regular"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  size: "regular", icon: "events", iconAlign: "left"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-left"
>
  <div
    className="Button-icon Button-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "regular", icon: "events", iconAlign: "right"
 1`] = `
<button
  className="Button Button--regular Button--basic Button--iconAlign-right"
>
  <div
    className="Button-icon Button-icon--right"
  >
    <Icon
      appearance="default"
      name="events"
      size={16}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "tiny"
 1`] = `
<button
  className="Button Button--tiny Button--basic Button--iconAlign-left"
>
  Button
</button>
`;

exports[`Button component 
  size: "tiny", icon: "events", iconAlign: "left"
 1`] = `
<button
  className="Button Button--tiny Button--basic Button--iconAlign-left"
>
  <div
    className="Button-icon Button-icon--left"
  >
    <Icon
      appearance="default"
      name="events"
      size={12}
    />
  </div>
  Button
</button>
`;

exports[`Button component 
  size: "tiny", icon: "events", iconAlign: "right"
 1`] = `
<button
  className="Button Button--tiny Button--basic Button--iconAlign-right"
>
  <div
    className="Button-icon Button-icon--right"
  >
    <Icon
      appearance="default"
      name="events"
      size={12}
    />
  </div>
  Button
</button>
`;

exports[`Button component with no children 
  appearance: "alert", loading: true
 1`] = `
<button
  className="Button Button--regular Button--square Button--alert"
  disabled={true}
>
  <Spinner
    appearance="white"
    size="small"
  />
</button>
`;

exports[`Button component with no children 
  appearance: "basic", loading: true
 1`] = `
<button
  className="Button Button--regular Button--square Button--basic"
  disabled={true}
>
  <Spinner
    appearance="secondary"
    size="small"
  />
</button>
`;

exports[`Button component with no children 
  appearance: "primary", loading: true
 1`] = `
<button
  className="Button Button--regular Button--square Button--primary"
  disabled={true}
>
  <Spinner
    appearance="white"
    size="small"
  />
</button>
`;

exports[`Button component with no children 
  appearance: "success", loading: true
 1`] = `
<button
  className="Button Button--regular Button--square Button--success"
  disabled={true}
>
  <Spinner
    appearance="white"
    size="small"
  />
</button>
`;

exports[`Button component with no children 
  appearance: "transparent", loading: true
 1`] = `
<button
  className="Button Button--regular Button--square Button--transparent"
  disabled={true}
>
  <Spinner
    appearance="secondary"
    size="small"
  />
</button>
`;
